src/rgw/rgw_rest_s3.cc

```diff
@@ -6377,14 +6564,22 @@ int RGWSelectObj_ObjStore_S3::send_response_data(bufferlist& bl, off_t ofs, off_
       continue; 
     }
 
+    m_aws_response_handler = std::make_unique<aws_response_handler>(s,this);
```

Comment:
this is wrong, m_aws_response_handler been initialized many times.
it also produces wrong results.
should do as follows
**_if (m_aws_response_handler.get() == nullptr)
{
m_aws_response_handler = std::make_unique<aws_response_handler>(s,this);
}_**